src/bootstrap/src/core/download.rs

```diff
@@ -21,6 +21,23 @@ fn try_run(config: &Config, cmd: &mut Command) -> Result<(), ()> {
     config.try_run(cmd)
 }
 
+fn extract_curl_version(out: &[u8]) -> f32 {
```

Comment:
We actually have the `semver` crate in dependencies, could you please try to parse the version using it? If it's not useful here, please at least also return the version using a tuple of integers.